### PR TITLE
#9 Add footer

### LINKED ISSUE #9: Create footer component
1. Create common project footer

### PR DESCRIPTION
**Pull request by OnlyR10**
Fixes #9

image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] Screenshot is attached or not needed

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
         "no-console": 0,
         "class-methods-use-this": "off",
         "@typescript-eslint/no-explicit-any": "error",
-        "linebreak-style": ["error", "windows"]
+        "linebreak-style": ["error", "windows"],
+        "import/extensions": ["error", { "tsx": "never" }]
```
Comment (nikolakozaryan) [inline .eslintrc.json]: This rule already exist in the config.

File: webpack.config.js [hunk 2]
```diff
@@ -59,7 +59,19 @@ module.exports = {
         rules: [
             {
                 test: /\.s[ac]ss$/i,
-                use: [MiniCssExtractPlugin.loader, 'css-loader', 'sass-loader'],
+                use: [
+                    MiniCssExtractPlugin.loader,
+                    {
```
Comment (nikolakozaryan) [inline webpack.config.js:76]: Don't understand why do we need this extension for the rule. Could you please explain?

File: src/style.scss [hunk 3]
```diff
@@ -1,3 +1,13 @@
 html {
```
Comment (nikolakozaryan) [inline src/style.scss]: Style.scss doesn't exist anymore in the develop branch. Base styles are in _base.scss file. Fonts connection is in the _typography.scss

File: src/index.tsx [hunk 4]
```diff
@@ -1,2 +1,8 @@
 import 'normalize.css';
 import './style.scss';
+import ReactDOM from 'react-dom/client';
+import App from './components/App';
+
+const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement);
```
Comment (nikolakozaryan) [inline src/index.tsx]: Also already exists.

File: src/assets/sass/abstract/mixins.scss [hunk 5]
```diff
@@ -0,0 +1,8 @@
+@mixin font($family: 'Gilroy', $style: normal, $weight: 400, $size, $height, $spacing: normal) {
```
Comment (nikolakozaryan) [inline src/assets/sass/abstract/mixins.scss]: I think we don't have so many font types to create mixin for this, just define neсessary font in the _variables.scss

**Comment by nikolakozaryan** [comment]
Please, add screenshots to each ui component when you open pr.

**Comment by nikolakozaryan** [comment]
Get the latest version of develop branch and fix structure misunderstanding please)

File: webpack.config.js (hunk 2 above)
Comment (OnlyR10) [reply webpack.config.js:76]: In order for class names in devTools to be readable, and not a random set of characters

File: .eslintrc.json (hunk 1 above)
Comment (OnlyR10) [reply .eslintrc.json]: A comma was added to this line, and there was no overwriting of the rule

Comment (nikolakozaryan) [reply .eslintrc.json]: Sorry))

File: webpack.config.js (hunk 2 above)
Comment (nikolakozaryan) [reply webpack.config.js:76]: Got it

File: src/components/footer/Footer.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+                <a href="https://github.com/" target="_blank" className={classes.iconGithub}></a>
+                <a href="https://github.com/nikolakozaryan" target="_blank" className={classes.linkToDeveloper}>
+                    Nikolay Kozyrev
+                </a>
+                <a href="https://github.com/OnlyR10" target="_blank" className={classes.linkToDeveloper}>
+                    Yakov Grigorev
+                </a>
+                <a href="https://github.com/YaGrig" target="_blank" className={classes.linkToDeveloper}>
+                    Kirill Naydovich
+                </a>
```
Comment (nikolakozaryan) [inline src/components/footer/Footer.tsx]: Mixed links to github. Swap them)

**Comment by OnlyR10** [comment]
> Please, add screenshots to each ui component when you open pr.

Done

**Comment by OnlyR10** [comment]
> Get the latest version of develop branch and fix structure misunderstanding please)

Done

File: src/App.tsx [hunk 7]
```diff
@@ -1,11 +1,12 @@
 import React from 'react';
 import 'normalize.css';
-import './common/style'
+import './sass/index.scss';
```
Comment (nikolakozaryan) [inline src/App.tsx]: There is no sass folder in the project.

File: src/assets/fonts/gilroy-medium.css [hunk 8]
```diff
@@ -0,0 +1,9 @@
+@font-face {
+	font-family: 'Gilroy-Medium';
+	src: url('../fonts/gilroy-medium.eot'); /* IE 9 Compatibility Mode */
+	src: url('../fonts/gilroy-medium.eot?#iefix') format('embedded-opentype'), /* IE < 9 */
+		url('../fonts/gilroy-medium.woff2') format('woff2'), /* Super Modern Browsers */
+		url('../fonts/gilroy-medium.woff') format('woff'), /* Firefox >= 3.6, any other modern browser */
+		url('../fonts/gilroy-medium.ttf') format('truetype'), /* Safari, Android, iOS */
+		url('../fonts/gilroy-medium.svg#gilroy-medium') format('svg'); /* Chrome < 4, Legacy iOS */
+}
```
Comment (nikolakozaryan) [inline src/assets/fonts/gilroy-medium.css]: Seems like fonts are already implemented in develop branch.

File: src/assets/fonts/gilroy-semibold.css [hunk 9]
```diff
@@ -0,0 +1,9 @@
+@font-face {
+	font-family: 'Gilroy-SemiBold';
+	src: url('../fonts/gilroy-semibold.eot'); /* IE 9 Compatibility Mode */
+	src: url('../fonts/gilroy-semibold.eot?#iefix') format('embedded-opentype'), /* IE < 9 */
+		url('../fonts/gilroy-semibold.woff2') format('woff2'), /* Super Modern Browsers */
+		url('../fonts/gilroy-semibold.woff') format('woff'), /* Firefox >= 3.6, any other modern browser */
+		url('../fonts/gilroy-semibold.ttf') format('truetype'), /* Safari, Android, iOS */
+		url('../fonts/gilroy-semibold.svg#gilroy-semibold') format('svg'); /* Chrome < 4, Legacy iOS */
+}
```
Comment (nikolakozaryan) [inline src/assets/fonts/gilroy-semibold.css]: The same.

File: src/components/button/Button.module.scss [hunk 10]
```diff
@@ -1,4 +1,4 @@
-@import '../../common/style/abstracts/variables';
+@import '../../sass/abstract/variables';
```
Comment (nikolakozaryan) [inline src/components/button/Button.module.scss]: The same problem (sass folder).

File: src/style.scss (hunk 3 above)
Comment (OnlyR10) [reply src/style.scss]: Done

File: src/index.tsx (hunk 4 above)
Comment (OnlyR10) [reply src/index.tsx]: Done

File: src/assets/sass/abstract/mixins.scss (hunk 5 above)
Comment (OnlyR10) [reply src/assets/sass/abstract/mixins.scss]: This gives you more control over the included styles.

File: src/components/footer/Footer.tsx (hunk 6 above)
Comment (OnlyR10) [reply src/components/footer/Footer.tsx]: Done

File: src/App.tsx (hunk 7 above)
Comment (OnlyR10) [reply src/App.tsx]: Done

File: src/assets/fonts/gilroy-medium.css (hunk 8 above)
Comment (OnlyR10) [reply src/assets/fonts/gilroy-medium.css]: The fonts have been extended by adding several formats

File: src/assets/fonts/gilroy-semibold.css (hunk 9 above)
Comment (OnlyR10) [reply src/assets/fonts/gilroy-semibold.css]: The fonts have been extended by adding several formats

File: src/components/button/Button.module.scss (hunk 10 above)
Comment (OnlyR10) [reply src/components/button/Button.module.scss]: Done

**Review by nikolakozaryan — APPROVED** [review]

**Pull request merged**